lib/core/src/Cardano/Wallet/Primitive/Types/UTxO.hs

```diff
@@ -127,6 +130,12 @@ restrictedTo :: UTxO -> Set TxOut -> UTxO
 restrictedTo (UTxO utxo) outs =
     UTxO $ Map.filter (`Set.member` outs) utxo
 
+null :: UTxO -> Bool
+null (UTxO u) = Map.null u
```

Comment:
Really? The only place this is used it's with `(not (UTxO.null x)`. But `x /= mempty` would be fine.

Reply:
> Really? The only place this is used it's with `(not (UTxO.null x)`. But `x /= mempty` would be fine.

I agree with you that `x /= mempty` would work fine!

Though I find that `UTxO.null` is a little more self-documenting (in this case), as:

- A reader doesn't have to navigate to the definition to figure out the type of thing being tested, because it's stated right at the point of use; and
- The `null` function is idiomatic across different container types, so it's familiar.